### ISSUE TITLE
HTSP v26 support

### ISSUE THREAD

**Issue by Glenn-1990** [kodi-pvr/pvr.hts#252]
Before I start coding stuff that won't make it in pvr.hts, I'll discuss my intentions for htsp v26 here. 

* playcount and playposition were added to tvheadend, we can introduce a setting for this (allow to choose between server or locally saved positions/counts) or let it depend on the htsp version automatically. 
* it's possible to log acces rights now, so if a user complains about function x or y is not working, we can easily check if that's do due access restrictions. https://github.com/tvheadend/tvheadend/commit/d6488b3077b07dd95aeb1a452632c2d51189403b
* Along with access rights, it's also possible to get the user interface level from tvheadend (basic, advanced or expert). It should be possible to adapt this interface level to the timersettings dialog. Example: UI level in tvheadend is "basic", this means that adding a recording with the webui does not ask for a storage path (uses default). Kodi can do the same, not showing the storage path. In this way Kodi and Tvheadend timer add/edit will be more correlated.
* "dvrEntryUpdate" will NOT send all properties every 5 seconds now when a recording is active. Only valid info will be transferred in order to limit traffic.

@ksooo @Jalle19

**Comment by Jalle19** [comment]
> playcount and playposition were added to tvheadend, we can introduce a setting for this (allow to choose between server or locally saved positions/counts) or let it depend on the htsp version automatically.

Personally I don't think we need a setting for this. Obviously people with existing setups would lose their watched status if we move to storing it on the server.

> it's possible to log acces rights now, so if a user complains about function x or y is not working, we can easily check if that's do due access restrictions. tvheadend/tvheadend@d6488b3

Not sure if we need this? @ksooo?

> Along with access rights, it's also possible to get the user interface level from tvheadend (basic, advanced or expert). It should be possible to adapt this interface level to the timersettings dialog. Example: UI level in tvheadend is "basic", this means that adding a recording with the webui does not ask for a storage path (uses default). Kodi can do the same, not showing the storage path. In this way Kodi and Tvheadend timer add/edit will be more correlated.

I think this would just cause confusion. People have trouble with Kodi's concept of setting visibility levels even to this day.

> "dvrEntryUpdate" will NOT send all properties every 5 seconds now when a recording is active. Only valid info will be transferred in order to limit traffic.

Sounds perfect!

tl;dr feel free to start implementing support

**Comment by ksooo** [comment]
>>    playcount and playposition were added to tvheadend, we can introduce a setting for this (allow to choose between server or locally saved positions/counts) or let it depend on the htsp version automatically.

> Personally I don't think we need a setting for this. Obviously people with existing setups would lose their watched status if we move to storing it on the server.

 No setting, yes.

 Regarding migration: good point. We need a way to detect whether we have once to transfer local play counts and resume points (and to do it, then), maybe also to delete local play counts/resume points after that.

>>    it's possible to log acces rights now, so if a user complains about function x or y is not working, we can easily check if that's do due access restrictions. tvheadend/tvheadend@d6488b3

> Not sure if we need this? @ksooo?

 What exactly would be the pvr.hts use case for this feature? You wrote: "we can easily check". Okay, but what to do then? Error dialog box or something like this?

>>    Along with access rights, it's also possible to get the user interface level from tvheadend (basic, advanced or expert). It should be possible to adapt this interface level to the timersettings dialog. Example: UI level in tvheadend is "basic", this means that adding a recording with the webui does not ask for a storage path (uses default). Kodi can do the same, not showing the storage path. In this way Kodi and Tvheadend timer add/edit will be more correlated.

> I think this would just cause confusion. People have trouble with Kodi's concept of setting visibility levels even to this day.

 Full ack. No need for this complexity. I hear users complaining "i was told i can do xyz in timer settings, but for me, xyz does not show up there. I will file a bug". Please, no.

>>    "dvrEntryUpdate" will NOT send all properties every 5 seconds now when a recording is active. Only valid info will be transferred in order to limit traffic.

> Sounds perfect!

 cool.

> feel free to start implementing support

 yes, please. I cant wait to get server-side play counts and resume points. ;-)
